packages/search-input/src/SearchResult/SearchResult.tsx

```diff
@@ -40,9 +40,10 @@ export const SearchResult = InferredPolymorphic<SearchResultProps, 'li'>(
       rest['aria-label'] ?? (rest['aria-labelledby'] ? '' : textContent);
 
     return (
+      // @ts-expect-error
       <InputOption
         {...rest}
-        as={as}
+        as={renderedAs}
```

Comment:
SInce we're just passing the `as` prop through, I believe we can just use the [Basic Polymorphic](https://github.com/mongodb/leafygreen-ui/blob/main/packages/polymorphic/README.md#basic-usage) usage

Reply:
It's not just passing thru, we're setting a default so we have to ask if `href` exists or not